find/find.go

```diff
@@ -0,0 +1,65 @@
+package find
+
+import (
+	"maps"
+	"slices"
+
+	kafka "github.com/confluentinc/confluent-kafka-go/v2/kafka"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"github.com/vordimous/gohlay/internal"
+	"github.com/vordimous/gohlay/kafkautil"
+)
+
+type Finder struct {
+	reason string
+	gohlayedMessages map[string]bool
+}
+
+// CheckForDeliveries will scan the topic and build a map of messages to be delivered
+func CheckForDeliveries() (*Finder) {
+	f := new(Finder)
+	f.reason = "indexing"
+	internal.ScanAll(f)
+	return f
+}
+
+
+// GetGohlayedSlice returns the map of gohlayed message keys
+func (f *Finder) GetGohlayed() map[string]bool {
+	return f.gohlayedMessages
+}
+
+// GetGohlayedSlice creates an array of strings from the gohlayed map keys
+func (f *Finder) GetGohlayedSlice() []string {
+	if d := slices.Collect(maps.Keys(f.GetGohlayed())); d != nil {
+		return d
+	}
+	return []string{}
+}
+
+func (f *Finder) GetReason() string {
+	return f.reason
+}
+
+func (f *Finder) HandleMessage(msg *kafka.Message) {
+	deadline := viper.GetInt64("deadline")
+	deliveryTime, delivered, deliveredKey, hasHeader := kafkautil.ParseHeaders(msg.Headers)
```

Comment:
I see quite a few of these where you return a host of values from a function. You could collect this in in a struct to improve readability somewhat. Also, you work a lot out of using bools, return them as errors instead to clean up logic between intended flow and edge cases
i.e
```
 kafkaHeaders, err := kafkautil.ParseHeaders(msg.Headers)
if err != nil {
   switch {
     case errors.Is(err, ErrNoHeader):
        ....
}
```

Reply:
addressing this in a new PR: https://github.com/vordimous/gohlay/issues/6